src/plugins/badge.coffee

```diff
@@ -0,0 +1,774 @@
+  </div>
+  """
+
+  TEMPLATE_EMBEDDED = (assigns) -> """
+  <div id="sa-badge-embedded-header">
+    <div id="sa-badge-embedded-logo"></div>
+    <div id="sa-badge-embedded-rating">
+      <div class="rating-number"><span>#{assigns.rating}</span></div>
+      <div class="stars-container">
+       #{(STAR_TEMPLATE_EMBEDDED(star_type, assigns) for star_type in assigns.stars).join("\n")}
+      </div>
+    </div>
+  </div>
+  <div id="sa-badge-embedded-footer">
+    <div id="sa-badge-embedded-more-button">@@translations.badge.more</div>
+  </div>
+  """
+
+  STAR_TEMPLATE_EMBEDDED = (type, assigns) -> """
+  <div class="sa-badge-star sa-badge-#{type}-star"></div>
+  """
+
+  STAR_TEMPLATE_FLOATING = (type, assigns) -> """
+  <div class="sa-badge-star sa-badge-#{assigns.theme}-theme-#{type}-star"></div>
+  """
+
+  MODAL_INNER_TEMPLATE = (assigns) -> """
+  <div id="sa-badge-modal-inner">
+    <span id="sa-badge-modal-close-button"></span>
+    <div id="sa-badge-modal-spinner" class="spinner"></div>
+    <iframe id="sa-badge-modal-iframe" src="#{assigns.endpoint}" width="100%" height="100%" frameborder="0"></iframe>
+    </div>
+  </div>
+  """
+
+  constructor: ->
+    @_setParentDoc()
+
+    return unless @_validate()
+
+    @addStyle()
+    @onDOMReady => @initialize()
+
+  _validate: ->
+    if context().configuration.display == 'embedded'
+      if not @$embeddedBadgePlugin()
+        @_log_error("could not find div[id='sa-badge-embedded-plugin']")
+        return false
+
+    true
+
+  initialize: ->
+    @render()
+    @_bindHandlers()
+
+  ###
+  Detect DOM ready
+  ###
+  onDOMReady: (cb) ->
+    # IE is only safe when readyState is 'complete'
+    return cb() if @parent_doc.readyState == 'complete'
+    return cb() if !@parent_doc.attachEvent && @parent_doc.readyState in ['interactive', 'complete']
+
+    if @parent_doc.addEventListener # Mozilla, Opera, Webkit
+      @parent_doc.addEventListener 'DOMContentLoaded', cb, false
+    else if @parent_doc.attachEvent # IE
+      @parent_doc.attachEvent('onreadystatechange', (=> cb() if @parent_doc.readyState == 'complete'))
+    else
+      setTimeout((=> try cb() catch), DOMREADY_TIMEOUT)
+
+  ###
+  Displays the plugin
+  ###
+  render: ->
+    switch context().configuration.display
+      when 'floating' then @_render_floating()
+      when 'embedded' then @_render_embedded()
+
+  _log_error: (message) ->
+    console.log("Skoutz Partner Badge: ERROR: " + message)
+
+  _render_floating: ->
+    @$el = document.createElement('div')
+    @$el.id = 'sa-badge-floating-plugin'
+    @$el.innerHTML = TEMPLATE_FLOATING(stars: @rating_to_stars(context().data.rating), theme: context().configuration.theme)
+    @parent_doc.body.appendChild(@$el)
+
+  _render_embedded: ->
+    @$el = @$embeddedBadgePlugin()
+    rating = context().data.rating
+    @$el.innerHTML = TEMPLATE_EMBEDDED(rating: rating, stars: @rating_to_stars(rating), embedded: true)
```

Comment:
No, it's not needed there as it's never used from `EMBEDDED_TEMPLATE`; theme is only used from `STYLE`. For example: 

```
  STYLE = (configuration) -> """
    ...
    #sa-badge-embedded-header #sa-badge-embedded-rating .stars-container .sa-badge-empty-star {
      background-image: url('#{asset_url("badge/embedded/theme/#{configuration.theme}/star_empty.png")}');
    }
    ...
  """
```

`theme` is also not needed anymore in the `FLOATING_TEMPLATE` call, so I'll remove it from there, too.